File: app/channels/timeline_channel.rb
class TimelineChannel < ApplicationCable::Channel
  def subscribed
    stream_from "timeline_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def like(data)
    idea = Idea.find(data["id"])
    idea.increment! :likes_count
    ActionCable.server.broadcast 'timeline_channel', { idea_id: idea.id, likes_count: idea.likes_count }
  end
end

File: app/javascript/channels/timeline_channel.js
import consumer from "./consumer"

export const timelineChannel = consumer.subscriptions.create("TimelineChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    const ideaCount = document.getElementById(`idea-count-${data["idea_id"]}`);
    ideaCount.textContent = data.likes_count
  },

  like: function(idea_id) {
    return this.perform('like', { id: idea_id });
  }
});

// document.addEventListener("turbolinks:load", () => {
//   const links = document.querySelectorAll('.likes-link');
//   links.forEach(link => {
//     link.addEventListener('click', e => {
//       timelineChannel.like(e.currentTarget.dataset.id);
//     });
//   })
// })
